### PR TITLE
AWS deployment update

### PR DESCRIPTION
**Pull request by tsimonyan**
- Upgrade to Terraform 0.12
- Remove all usage of nginx. Use Cerebro and Grafana's own webservers and authentication
- Elasticsearch and Kibana auth now uses XPack Security. Self-signed certs for internode connection are managed by Terraform
- Master/Data nodes now automatically mount their EBS disks. Nodes can be terminated without losing state when brought back up
- Change Classic load balancer with ALB
- Move user_data scripts to Packer stage

### CODE REVIEW

File: packer/install-beats.sh [hunk 1]
```diff
@@ -1,16 +1,14 @@
...
+curl -L -O https://artifacts.elastic.co/downloads/beats/filebeat/filebeat-7.5.0-amd64.deb
+sudo dpkg -i filebeat-7.5.0-amd64.deb
+rm filebeat-7.5.0-amd64.deb
 
-curl -L -O https://artifacts.elastic.co/downloads/beats/filebeat/filebeat-6.4.3-amd64.deb
-sudo dpkg -i filebeat-6.4.3-amd64.deb
-rm filebeat-6.4.3-amd64.deb
+curl -L -O https://artifacts.elastic.co/downloads/beats/heartbeat/heartbeat-7.5.0-amd64.deb
+sudo dpkg -i heartbeat-7.5.0-amd64.deb
+rm heartbeat-7.5.0-amd64.deb
```
Comment (synhershko) [inline packer/install-beats.sh:10]: use es_version variable passed to packer if exists, and if it didn't - is there a way to determine latest beats version and default to it?

File: packer/update-machine.sh [hunk 2]
```diff
@@ -2,15 +2,20 @@
...
 # Disable daily apt unattended updates.
 echo 'APT::Periodic::Enable "0";' >> /etc/apt/apt.conf.d/10periodic
 
-wget https://bootstrap.pypa.io/get-pip.py
-python get-pip.py || python2 get-pip.py
-pip install boto awscli || pip2 install boto awscli 
+cd /tmp
+curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "/tmp/awscliv2.zip"
+unzip /tmp/awscliv2.zip -d /tmp
+sudo /tmp/aws/install
```
Comment (synhershko) [inline packer/update-machine.sh:21]: Ubuntu has aws-cli package that can be installed via apt-get - can we use that instead?

File: packer/install-beats.sh (hunk 1 above)
Comment (tsimonyan) [reply packer/install-beats.sh:10]: We can get the latest ES version using Bintray API. Assuming there is a beats release for each ES version.

Comment (synhershko) [reply packer/install-beats.sh:10]: Let's use the es_version env var if exist and fallback to manually updated latest version for now

File: terraform-aws/variables.tf [hunk 3]
```diff
@@ -68,73 +62,69 @@ variable "elasticsearch_logs_dir" {
...
 
 variable "client_heap_size" {
-  type = "string"
+  type    = string
   default = "1g"
 }
 
 variable "masters_count" {
-  default = "0"
+  type        = map(number)
```
Comment (tyler-ground) [inline terraform-aws/variables.tf:80]: why is this a map?  What would be the masters_count for 3 masters?

**Comment by jederfreitas** [comment]
Will it be  updated to azure as well?

**Comment by synhershko** [comment]
Hi @jederfreitas , we don't currently have active customers that need it so we don't have plans to update the Azure deployment soon. If that changes we will test and update the Azure deployment as well.

**Pull request merged**